### PR TITLE
Release 0.12

### PR DESCRIPTION
**Pull request by zrzka**
List of crates to release (ordered):

- [x] crossterm_winapi - 0.3.0
- [x] crossterm_utils - 0.4.0
- [x] crossterm_screen - 0.3.2
- [x] crossterm_style - 0.5.2
- [x] crossterm_input - 0.5.0
- [x] crossterm_cursor - 0.4.0
- [x] crossterm_terminal - 0.3.2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 
 - Deprecated warning about `crossterm_*` crates
 - Documentation improved
 - Remove all references to the crossterm book
 - Remove the crossterm book
+- New commands introduced
+  - `EnterAlternateScreen`
+  - `LeaveAlternateScreen`
+  - `ResetColor`
+- Fixed bug with `SetBg` command, WinApi logic
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-style/pull/5

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+- Fixed bug with `StyledObject`, used stdout for resetting terminal color
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-style/pull/2

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+- Mouse coordinates synchronized with the cursor (breaking)
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-input/pull/7

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+  - Upper/left reported as `(0, 0)`
+- Fixed bug that read sync didn't block (Windows)
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-winapi/pull/2

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,9 +1,22 @@
...
+- AsyncReader produces mouse events (#271)
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-input/pull/9

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,9 +1,22 @@
...
+  - One reading thread per application, not per AsyncReader
+- Fixed cursor position getting consumed by async reader
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-input/pull/9

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,9 +1,22 @@
...
+- Implemented sync reader for read_char (requires raw mode)
```
Comment (TimonPost) [inline CHANGELOG.md]: https://github.com/crossterm-rs/crossterm-input/pull/9

Comment (TimonPost) [reply CHANGELOG.md]: Also: "removed all unsafe static code"

**Comment by zrzka** [comment]
Is it worth it? I mean, I'm going to add PR links to all sub-crates, but here? I mean one changelog line will lead to multiple PRs, ... - for example this static mut change.

**Comment by zrzka** [comment]
Changelogs in all PRs updated. Changelog of the `crossterm` crate contains verbatim copy of all these changelogs in separate sections.

**Pull request merged**